### PR TITLE
Simplify some JUnit related things

### PR DESCRIPTION
**Pull request by gzm0**
This is to prepare for adding async support to JUnit.

### CODE REVIEW

File: junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala [hunk 1]
```diff
@@ -113,386 +55,170 @@ class ScalaJSJUnitPlugin(val global: Global) extends NscPlugin {
...
-       */
-      private def genNewInstanceDef(classSym: Symbol, bootSymbol: Symbol): DefDef = {
-        val mkNewInstanceDefRhs =
-          mkNewInstance(TypeTree(classSym.typeConstructor), Nil)
-        val mkNewInstanceDefSym = bootSymbol.newMethodSymbol(newTermName("newInstance"))
-        mkNewInstanceDefSym.setInfo(MethodType(Nil, definitions.ObjectTpe))
+        val rhs = tests.foldRight[Tree] {
+          Throw(New(typeOf[NoSuchMethodException], Ident(nameParam)))
+        } { (sym, next) =>
+          val cond = Apply(
```
Comment (sjrd) [inline junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala]: `gen.mkMethodCall`?

File: junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala [hunk 2]
```diff
@@ -113,386 +55,170 @@ class ScalaJSJUnitPlugin(val global: Global) extends NscPlugin {
...
+              Select(Ident(nameParam), definitions.Object_equals),
+              List(Literal(Constant(sym.name.toString))))
 
-        typer.typedDefDef(newDefDef(mkNewInstanceDefSym, mkNewInstanceDefRhs)())
-      }
+          val call = Apply(Select(castParam(instanceParam, testClass), sym), Nil)
```
Comment (sjrd) [inline junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala]: `gen.mkMethodCall`?

File: junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala [hunk 3]
```diff
@@ -113,386 +55,170 @@ class ScalaJSJUnitPlugin(val global: Global) extends NscPlugin {
...
-        val tree = methods.foldRight[Tree](mkMethodNotFound(idParamSym)) { (methodSymbol, acc) =>
-            val mName = Literal(Constant(methodSymbol.name.toString))
-            val paramIdent = gen.mkAttributedIdent(idParamSym)
-            val cond = gen.mkMethodCall(paramIdent, definitions.Object_equals, Nil, List(mName))
-            val call = genCall(methodSymbol)
-            If(cond, call, acc)
-        }
-        atOwner(methodSym)(typer.typed(tree))
+      private def genNewInstance(owner: ClassSymbol, testClass: ClassSymbol): DefDef = {
+        val sym = owner.newMethodSymbol(newTermName("newInstance"))
```
Comment (sjrd) [inline junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala]: Another `TermName`

File: junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala [hunk 4]
```diff
@@ -113,386 +55,170 @@ class ScalaJSJUnitPlugin(val global: Global) extends NscPlugin {
...
-      private def mkMethodNotFound(paramSym: Symbol) = {
-        val paramIdent = gen.mkAttributedIdent(paramSym)
-        val msg = gen.mkMethodCall(paramIdent, definitions.String_+, Nil,
-          List(Literal(Constant(" not found"))))
-        val exception = mkNewInstance[NoSuchMethodException](List(msg))
-        Throw(exception)
-      }
-    }
+      private def castParam(param: Symbol, clazz: Symbol): Tree =
+        TypeApply(Select(Ident(param), definitions.Object_asInstanceOf), List(TypeTree(clazz.tpe)))
```
Comment (sjrd) [inline junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala]: `gen.mkAttributedCast`

File: junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala [hunk 5]
```diff
@@ -113,386 +55,170 @@ class ScalaJSJUnitPlugin(val global: Global) extends NscPlugin {
...
 
-    private lazy val hasNewCollections = {
-      val v = scala.util.Properties.versionNumberString
-      !v.startsWith("2.10.") &&
-      !v.startsWith("2.11.") &&
-      !v.startsWith("2.12.") &&
-      v != "2.13.0-M3"
+      private def annotatedMethods(owner: Symbol, annot: Symbol): Scope =
+        owner.info.members.filter(m => !m.isBridge && m.hasAnnotation(annot))
```
Comment (sjrd) [inline junit-plugin/src/main/scala/org/scalajs/junit/plugin/ScalaJSJUnitPlugin.scala]: Should this have `m.isMethod`?

**Comment by gzm0** [comment]
Updated.

**Pull request merged**